etc/profile-a-l/com.github.tchx84.Flatseal.profile

```diff
@@ -0,0 +1,57 @@
+# Firejail profile for flatseal
+# This file is overwritten after every install/update
+# Persistent local customizations
+include com.github.tchx84XXgg12uC.Flatseal.local
+# Persistent global definitions
+include globals.local
+
+# Required for GTK theem to work:
+read-only ${HOME}/.config/dconf/
+noblacklist ${HOME}/.config/dconf/
+whitelist ${HOME}/.config/dconf/
+
+# TODO: Should be relative to xdg-data-home
+mkdir ${HOME}/.local/share/flatpak/overrides
+noblacklist ${HOME}/.local/share/flatpak/overrides
+whitelist ${HOME}/.local/share/flatpak/overrides
+
+include disable-devel.inc
+include disable-exec.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-shell.inc
+include disable-write-mnt.inc
+include disable-xdg.inc
+
```

Comment:
What about wruc? wrc? wusc? wvc?

Reply:
If you don't include them, you grant full access (instead of restricted access). Include them if they work (wusc maybe need `whitelist /usr/share/flatseal`).

Reply:
_`s|/usr|/usr/share|`_

Yes, that's how whitelisting works.
You can try it with wusc included and not included: `firejail --ignore=private-bin --profile=/path/to/flatseal.profile ls /usr/share`

Reply:
Oh, so access to `/usr/share` is unrestricted by default, but if anything inside that directory is whitelisted, then _only_ the whitelisted directory is visible, and everything else is hidden.

Right, re-reading the man page now, that's actually what is says, but I somehow didn't quite understand that the first time, thanks for the clarification.

Reply:
Updated this bit.

Honestly, I think the whitelisting mechanic is a bit hard to fully understand. On the other hand, I can't think of any obvious way of making it simpler, so don't take this as a hard criticism.

Reply:
`include whitelist-run-common.inc`?

Reply:
> Honestly, I think the whitelisting mechanic is a bit hard to fully understand. On the other hand, I can't think of any obvious way of making it simpler, so don't take this as a hard criticism.

Yes, it's probably the hardest one (if we ignore this like seccomp which go deeper in how linux works). Based on the discussion in #3447, the idea is to have `mask /usr/share` to enable/disable this behaviour (if anyone ever implements this).

Reply:
The thing that doesn't fully convince me about the whitelist mechanic, is that by default, all access to `/run` is allowed, so we use `include whitelist-run-common.inc`, but that still grants access to various unnecessary items (including printing!?) which this application doesn't **need**.

Ideally, `/run` should be entirely hidden here.

Reply:
> including printing?

Then add `noprinters`

> Ideally, /run should be entirely hidden here.

Not possible `/run/user` is required and `/run/firejail` too.

>  but that still grants access to various unnecessary items which this application doesn't need.

In specific of wrc, see #4607.

Reply:
Added `noprinters`.